lib/tst_res.c

```diff
@@ -81,8 +81,13 @@ int TEST_ERRNO;
 	assert(strlen(buf) > 0);		\
 } while (0)
 
-#if defined(__ANDROID__) && !defined(PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP)
+#ifndef PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP
+#ifdef __ANDROID__
 #define PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP PTHREAD_RECURSIVE_MUTEX_INITIALIZER;
```

Comment:
this ends in a `;` while the one below does not.  we probably should drop the `;` here.

Reply:
Yes, we should. But in a separate commit. This comes from 34a7c82a7 ("android: exclude tests which don't build"), one of the initial commits fixing android build.

Reply:
@vapier is correct that without this change, this will "fail" a "checkpatch" with a warning, but based on @pevik suggestion had kept it AS-IS and made the fix for android in a second commit.

it would be better if someone could at least test build it in android before it gets merged though.